presto-spi/src/main/java/com/facebook/presto/spi/plan/MergeJoinNode.java

```diff
@@ -82,6 +83,15 @@ public MergeJoinNode(
         this.rightHashVariable = requireNonNull(rightHashVariable, "rightHashVariable is null");
     }
 
+    /**
+     * @param joinNode
+     * @return returns true if merge join is supported for the given join node.
+     */
+    public static boolean isMergeJoinEligible(JoinNode joinNode)
+    {
+        return joinNode.getType() == INNER && !joinNode.isCrossJoin();
```

Comment:
This condition can be enhanced to include `LEFT` and `RIGHT` joins as well.

Reply:
ditto